online_meetings/canvas_zoom_meetings.py

```diff
@@ -0,0 +1,138 @@
+
+import pandas as pd
+
+# read configurations
+try:
+    with open(os.path.join('../config', 'env.json')) as env_file:
+        ENV = yaml.load(env_file.read())
+except FileNotFoundError:
+    sys.exit(
+        'Configuration file could not be found; please add env.json to the config directory.')
+
+LOG_LEVEL = ENV.get('LOG_LEVEL', 'DEBUG')
+logging.basicConfig(level=LOG_LEVEL)
+
+# These two lines enable debugging at httplib level (requests->urllib3->http.client)
+# You will see the REQUEST, including HEADERS and DATA, and RESPONSE with HEADERS but without DATA.
+# The only thing missing will be the response.body which is not logged.
+if LOG_LEVEL == logging.DEBUG:
+    http.client.HTTPConnection.debuglevel = 1
+
+# You must initialize logging, otherwise you'll not see debug output.
+logger = logging.getLogger(__name__)
+logger.setLevel(LOG_LEVEL)
+
+requests_log = logging.getLogger("requests.packages.urllib3")
+requests_log.setLevel(LOG_LEVEL)
+requests_log.propagate = True
+# Global variable to access Canvas
+CANVAS = Canvas(ENV.get("CANVAS_URL"), ENV.get("CANVAS_TOKEN"))
+
+
+def zoom_course_report(canvas_account=1, enrollment_term_id=1, published=True):
+    zoom_courses = []
+    zoom_courses_meetings = []
+
+    account = CANVAS.get_account(canvas_account)
+    # Canvas has a limit of 100 per page on this API
+    per_page = 100
+    # Get all published courses from the defined enrollment term
+    courses = account.get_courses(enrollment_term_id=enrollment_term_id, published=published, per_page=per_page)
+    course_count = 0
+    for course in courses:
+        course_count += 1
+        # TODO: In the future get the total count from the Paginated object
+        # Needs API support https://github.com/ucfopen/canvasapi/issues/114
+        logger.info(f"Fetching course #{course_count} for {course}")
+        # Get tabs and look for zoom
+        tabs = course.get_tabs()
+        for tab in tabs:
+            # Hidden only included if true
+            if (tab.label == "Zoom" and not hasattr(tab, "hidden")):
+                logger.info("Found a course with zoom as %s", tab.id)
+                r = CANVAS._Canvas__requester.request("GET", _url=tab.url)
+                external_url = r.json().get("url")
+                r = requests.get(external_url)
+                # Parse out the form from the response
+                soup = bs(r.text, 'html.parser')
+                # Get the form and parse out all of the inputs
+                form = soup.find('form')
+                if not form:
+                    logger.info("Could not find a form to launch this zoom page, skipping")
+                    break
+
+                fields = form.findAll('input')
+                formdata = dict((field.get('name'), field.get('value')) for field in fields)
+
+                # Get the URL to post back to
+                posturl = form.get('action')
+
+                # Start up the zoom session
+                zoom_s = requests.Session()
```

Comment:
can we use `zoom_session` instead?

Reply:
Yeah this was me being lazy I think, I can do this.